enum.rb

```diff
@@ -0,0 +1,124 @@
+# frozen_string_literal: true
+
+# Microverse-202-Advanced-Building-Blocks-Enumerables
+# By Jair Jaramillo
+
+# module enum
+module Enumerable
+  def bubble_sort(array)
+    (array.size - 1).times do |x|
+      array[x], array[x + 1] = array[x + 1], array[x] if array[x] > array[x + 1]
+    end
+    (array.size - 1).times do |x|
+      bubble_sort(array) if array[x] > array[x + 1]
+    end
+    array
+  end
+
+  def bubble_sort_by(arra)
+    (arra.size - 1).times do |x|
+      arra[x], arra[x + 1] = arra[x + 1], arra[x] if yield(arra[x], arra[x + 1]).positive?
+    end
+    (arra.size - 1).times do |x|
+      next unless (arra[x].length - arra[x + 1].length).positive?
+
+      bubble_sort_by(arra) do |_left, _right|
+        yield(arra[x], arra[x + 1])
+      end
+    end
+    arra
+  end
+
+  def my_each(counter = 0)
+    while counter < to_a.length
+      yield to_a[counter]
```

Comment:
- [ ] my_each returns an Enumerator if no block is given
- [ ] my_each_with_index returns an enumerator if no block is given